### PR TITLE
Cleanup release python scripts

### PR DESCRIPTION
**Pull request by james-d-mitchell**
This PR applies some (minor) clean ups to the release python scripts in `dev/release`, there shouldn't be any functional changes, just updates to the formatting, and other issues highlighted by linting.

### CODE REVIEW

**Review by fingolfin — APPROVED** [review]
Thanks!

**Comment by fingolfin** [comment]
Needs to be rebased (sorry) afterwards let's just merge it.

I guess at some point we should add a CI test that runs `black` to verify the files are still formatted in each PR. We have something like that in the PackageDistro repository

**Comment by fingolfin** [comment]
@james-d-mitchell please ping me when you had time to update this, so I don't forget to merge it :-)

**Comment by james-d-mitchell** [comment]
@fingolfin I've updated this PR

File: dev/releases/update_website.py [hunk 1]
```diff
@@ -80,21 +80,21 @@ def download_asset_by_name(asset_name: str, writedir: str) -> None:
         utils.download_with_sha256(url, asset_name)
 
 
-def extract_tarball(tarball: str) -> None:
```
Comment (fingolfin) [inline dev/releases/update_website.py:83]: Why is `tarball` renamed to `tball`? (No objection, just wondering what the rationale is)

File: dev/releases/update_website.py [hunk 2]
```diff
@@ -80,21 +80,21 @@ def download_asset_by_name(asset_name: str, writedir: str) -> None:
...
 
 
 def get_date_from_configure_ac(gaproot: str) -> str:
-    with open(f"{gaproot}/configure.ac", "r") as configure_ac:
+    with open(f"{gaproot}/configure.ac", "r", encoding="utf-8") as configure_ac:
         filedata = configure_ac.read()
         try:  # Expect date in YYYY-MM-DD format
             release_date = re.search(
-                "\[gap_releaseday\], \[(\d{4}-\d{2}-\d{2})\]", filedata
+                r"\[gap_releaseday\], \[(\d{4}-\d{2}-\d{2})\]", filedata
```
Comment (fingolfin) [inline dev/releases/update_website.py:97]: Huh. Was this broken before then? Or is this just recommended but it works the same anyway?

If this changes behavior we should check what the effect is, if any.

**Review by fingolfin — APPROVED** [review]
Thanks!

File: dev/releases/update_website.py (hunk 1 above)
Comment (james-d-mitchell) [reply dev/releases/update_website.py:83]: Because `tarball` is a variable defined in an outer scope, according to pylint.

**Pull request merged**